### PR TITLE
zssh: call autoreconf -fvi before configure, this fixes build issues …

### PR DESCRIPTION
**Pull request by areese**
…on m1.  Add a simple test

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
> tcgetattr: Inappropriate ioctl for device

**Comment by areese** [comment]
> tcgetattr: Inappropriate ioctl for device

@SMillerDev lemme check that again.  I thought I stopped seeing that.

**Comment by areese** [comment]
```
+ pkg=zssh
+ brew reinstall zssh --build-from-source
==> Downloading https://downloads.sourceforge.net/project/zssh/zssh/1.5/zssh-1.5c.tgz
Already downloaded: /Users/allenreese/Library/Caches/Homebrew/downloads/2e67d64bb6f4034be2a728a8961f40087b3d878c3bd0af663f2af905b6e70125--zssh-1.5c.tgz
==> Reinstalling zssh 
==> autoreconf -fvi
==> ./configure --prefix=/opt/homebrew/Cellar/zssh/1.5c
==> make
🍺  /opt/homebrew/Cellar/zssh/1.5c: 10 files, 104.4KB, built in 7 seconds
+ brew test zssh
==> Testing zssh
==> /opt/homebrew/Cellar/zssh/1.5c/bin/zssh -V
+ brew audit --strict zssh
zssh:
  * Formula zssh contains deprecated SPDX licenses: ["GPL-2.0"].
    You may need to add `-only` or `-or-later` for GNU licenses (e.g. `GPL`, `LGPL`, `AGPL`, `GFDL`).
    For a list of valid licenses check: https://spdx.org/licenses/
Error: 1 problem in 1 formula detected
SpeedyMolasses Formula % zssh
Press ^@ (C-Space) to enter file transfer mode, then ? for help

usage: ssh [-46AaCfGgKkMNnqsTtVvXxYy] [-B bind_interface]
           [-b bind_address] [-c cipher_spec] [-D [bind_address:]port]
           [-E log_file] [-e escape_char] [-F configfile] [-I pkcs11]
           [-i identity_file] [-J [user@]host[:port]] [-L address]
           [-l login_name] [-m mac_spec] [-O ctl_cmd] [-o option] [-p port]
           [-Q query_option] [-R address] [-S ctl_path] [-W host:port]
           [-w local_tun[:remote_tun]] destination [command]
```

I wonder if this happens on older MacOS?  @carlocab and I chased down a weird error for some time, lemme see if I can get an 11.3 installed.

**Comment by carlocab** [comment]
Might be able to work around that error by simulating a pty. I think it should be something like `require "pty"` and `PTY.spawn`.

**Comment by areese** [comment]
Added PTY.spawn, let's see if it passes.

**Comment by carlocab** [comment]
It works 🎉

**Comment by areese** [comment]
@carlocab good call. I wouldn't have figured that out cause my terminal lol.
maybe I should be doing `ssh localhost brew test` lols

File: Formula/zssh.rb [hunk 1]
```diff
@@ -27,11 +30,17 @@ class Zssh < Formula
   end
 
   def install
+    system "autoreconf", "-fvi"
```
Comment (SMillerDev) [inline Formula/zssh.rb:33]: Is this only for arm?

Comment (areese) [reply Formula/zssh.rb:33]: Yes.  it was passing on intel.  lemme add the if arm

File: Formula/zssh.rb [hunk 2]
```diff
@@ -19,6 +19,9 @@ class Zssh < Formula
     sha256 cellar: :any_skip_relocation, yosemite:    "94280569f9e1c1deb9d8c3be4256cd501399fd51758f8e2ea6d77fd9f1b6ef2e"
   end
 
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool"  => :build
```
Comment (carlocab) [inline Formula/zssh.rb:24]: Since we're only running `autoreconf` on ARM, these deps should be scoped inside an `if Hardware::CPU.arm?` block as well.

Comment (areese) [reply Formula/zssh.rb:24]: will do.

Comment (areese) [reply Formula/zssh.rb:24]: done

**Comment by areese** [comment]
ugh forgot to run audit before push.  :(

**Review by carlocab — DISMISSED** [review]

**Comment by carlocab** [comment]
Failed to build on 10.15 and 11-x86_64. Might actually need `autoreconf` there too. Will restart CI to double check.

**Comment by areese** [comment]
@carlocab I'll add the autoreconf back to see if it fixes 11/10.15

**Comment by carlocab** [comment]
We should try to find out why `autoreconf` is needed now on Catalina and Big Sur when it wasn't before. Might be hard, though, as I believe the Catalina bottle was build in this commit: https://github.com/Homebrew/homebrew-core/commit/0e9924a4cdf2fa8107d5930359997ff51c0ea90e

**Comment by areese** [comment]
I can give that a shot.

**Comment by carlocab** [comment]
Actually, that's only Catalina. We never managed to build this for Big Sur, so nothing changed on that front.

**Comment by areese** [comment]
> Actually, that's only Catalina. We never managed to build this for Big Sur, so nothing changed on that front.

Oh that makes sense
I tried Big Sur and was confused as to why things broke

**Comment by areese** [comment]
@carlocab so if I do the autoreconf this builds on Big Sur on intel and Apple Silicon.  (11.4)

**Review by carlocab — APPROVED** [review]

**Pull request merged**